File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        if(!list1 && !list2){
            return 
        }
        let curr1 = list1
        let curr2 = list2
        let result = new ListNode()
        result.val = null
        let currResult = result
        while (curr1 || curr2){
            console.log(curr1.val, curr2.val)
            if(curr1.val<=curr2.val){
                currResult.val = curr2.val
                currResult.next = new ListNode()
            }
            else{
                currResult.val = curr2.val
                currResult.next = new ListNode()
            }
            
            curr1 = curr1.next
            
            curr2 = curr2.next
            }
            console.log(result)
            return result
        }
    }

File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        
        fast = head
        slow = head

        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
        

        curr = slow.next
        slow.next = None
        prev = None

        while curr:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp

        first, second = head, prev

        while second:
            temp1, temp2 = first.next, second.next

            first.next = second
            second.next = temp1
            first, second = temp1, temp2





